cds-snc/forms-terraform
# fix: rework response archiver lambda

# Summary | Résumé

- Reworked response archiver lambda.
  - Increased lambda timeout from 3 seconds to 5 minutes
  - Code now processes chunks of 100 responses at a time
  - Slightly changed archiving location for both the submission and potential attachments to group them so that it is easier to reassemble 
  
<img width="1284" alt="Screenshot 2024-01-12 at 2 08 52 PM" src="https://github.com/cds-snc/forms-terraform/assets/1763961/88d52005-5dd0-46ce-83c7-b37d3cda9684">

# Test instructions

- Create form with file inputs (you have to turn on experimental features in the feature flags). Having file inputs allows us to fully test the archiving lambda but it can work without it as well.
- Submit at least one response to that form you created.
- Manually edit the `removalDate` of your response(s) using the https://app.localstack.cloud/inst/default/resources tool. Changed it so that the timestamp is in the past.
- Manually trigger the response archiver lambda using `awslocal lambda invoke --function-name Response_Archiver /dev/null`.
- The response(s) should now be archived. You can check both the DynamoDB and S3 services to see the result. The response should have been removed from DynamoDB and saved in the `forms-local-archive-storage ` S3 bucket.
